Packs/CiscoWSA/Integrations/CiscoWSAV2/CiscoWSAV2_test.py

```diff
@@ -0,0 +1,1051 @@
+    When:
+    - cisco-wsa-identification-profiles-delete command called.
+    Then:
+    - Ensure readable output is correct.
+    """
+    from CiscoWSAV2 import delete_identification_profiles_command
+
+    url = f"{BASE_URL}/{V3_PREFIX}/web_security/identification_profiles"
+    requests_mock.delete(url=url, status_code=HTTPStatus.NO_CONTENT)
+
+    result = delete_identification_profiles_command(
+        mock_client,
+        {
+            "profile_names": "test",
+        },
+    )
+
+    assert result.readable_output == "Deleted identification profiles successfully."
+
+
+def test_delete_fail_identification_profiles_command(
+    requests_mock,
+    mock_client,
+):
+    """
+    Scenario: Identification profile delete.
+    Given:
+    - User provided wrong arguments.
+    When:
+    - cisco-wsa-identification-profiles-delete command called.
+    Then:
+    - Ensure readable output is correct.
+    """
+    from CiscoWSAV2 import delete_identification_profiles_command
+
+    mock_response = load_mock_response("identification_profiles_delete_fail.json")
+
+    url = f"{BASE_URL}/{V3_PREFIX}/web_security/identification_profiles"
+    requests_mock.delete(url=url, status_code=HTTPStatus.MULTI_STATUS, json=mock_response)
+
+    result = delete_identification_profiles_command(
+        mock_client,
+        {
+            "profile_names": "test,test2,test3",
+        },
+    )
+
+    assert len(result) == 3
+
+
+def test_list_url_categories_command(
+    requests_mock,
+    mock_client,
+):
+    """
+    Scenario: URL categories list.
+    Given:
+    - User provided valid credentials.
+    - User may provided pagination args.
+    - User may Provided filtering arguments.
+    When:
+    - cisco-wsa-url-categories-list command called.
+    Then:
+    - Ensure outputs prefix is correct.
+    - Ensure number of items is correct.
+    - Validate outputs' fields.
+    """
+    from CiscoWSAV2 import list_url_categories_command
+
+    mock_response = load_mock_response("url_categories_list.json")
+    url = f"{BASE_URL}/{V3_PREFIX}/generic_resources/url_categories"
+    requests_mock.get(url=url, json=mock_response)
+
+    result = list_url_categories_command(mock_client, {})
+
+    assert result.outputs_prefix == "CiscoWSA.UrlCategory"
+    assert len(result.outputs["predefined"]) == 106
+    assert len(result.outputs["custom"]) == 1
+
+
+""" TESTING HELPER FUNCTIONS"""
+
+
+@pytest.mark.parametrize(
+    "response,arguments,paginated_response",
+    [
+        (
+            ["test.com", "test1.com", "test2.com", "test3.com", "test4.com"],
+            {
+                "page": 2,
+                "page_size": 2,
+            },
+            ["test2.com", "test3.com"],
+        ),
+        (
+            ["test.com", "test1.com", "test2.com", "test3.com", "test4.com"],
+            {
+                "limit": 3,
+            },
+            ["test.com", "test1.com", "test2.com"],
+        ),
+    ],
+)
```

Comment:
What about in case the user provides all pagination arguments (page, page_size, and limit)?
I think you should handle this in the integration code and also test this here.